Source/Extensions/Blazorise.LoadingIndicator/LoadingIndicatorService.cs

```diff
@@ -0,0 +1,196 @@
+﻿#region Using directives
+using System;
+using System.Collections.Generic;
+using System.Linq;
+#endregion
+
+namespace Blazorise.LoadingIndicator
+{
+    /// <summary>
+    /// A service to control LoadingIndicator components
+    /// </summary>
+    public class LoadingIndicatorService: ILoadingIndicatorService
+    {
+        #region Members
+
+        private object hashLock = new();
+        private HashSet<LoadingIndicator> indicators = new();
+
+        // avoid locking in single indicator (app busy) scenario
+        Action<bool> SetBusyAction;
+        Action<bool> SetLoadedAction;
+        Func<bool?> GetBusyFunc;
+        Func<bool?> GetLoadedFunc;
+
+        #endregion
+
+        #region Methods
+
+        /// <inheritdoc />
+        public LoadingIndicatorService()
+        {
+            // default to foreach implementation
+            MultiMode();
+        }
+
+        // use locking implementation
+        private void MultiMode()
+        {
+            SetBusyAction = SetBusyMulti;
+            SetLoadedAction = SetLoadedMulti;
+            GetBusyFunc = GetBusyMulti;
+            GetLoadedFunc = GetLoadedMulti;
+        }
+
+        // no lock implementation
+        private void SingleMode( LoadingIndicator indicator )
+        {
+            SetBusyAction = indicator.SetBusy;
+            SetLoadedAction = indicator.SetLoaded;
+            GetBusyFunc = () => indicator.Busy;
+            GetLoadedFunc = () => indicator.Loaded;
+        }
+
+        /// <summary>
+        /// Show loading indicator
+        /// </summary>
+        public void Show() => SetBusyAction( true );
```

Comment:
`Show`, `Hide`, `SetLoaded` should all be `Task`-able. Long term this is better because all our other show/hide APIs are awaitable.